TELCODOCS-1506 Telco CORE Reference Design Specification (#65732)
openshift/openshift-docs: telco_ref_design_specs/core/telco-core-rds-use-cases.adoc

:_content-type: ASSEMBLY
[id="telco-ran-rds-overview"]
= Telco core use case model overview
include::_attributes/common-attributes.adoc[]
:context: ran-core-design-overview

toc::[]

> The use model for a telco core cluster provides a platform supporting large scale telecom applications including control plane functions like signaling and aggregation, and some centralized data plane for example user plane functions (UPF) functions running on commodity hardware. For examples of these functions see the architecture diagram. These functions generally require scalability, complex networking support, resilient software-defined storage, and support performance requirements that are less stringent and constrained than far-edge deployments like RAN.

Reviewer: ```suggestion
The use model for a telco core cluster provides a platform supporting large-scale telco applications including control plane functions like signaling and aggregation, and some centralized data plane functions. For example, user plane functions (UPF) running on commodity hardware. These functions generally require scalability, complex networking support, resilient software-defined storage, and support performance requirements that are less stringent and constrained than far-edge deployments like RAN.
```

"For examples of these functions see the architecture diagram." There is no diagram?

> The networking prerequisites for core telco functions exhibit large diversity, encompassing an array of networking attributes and performance benchmarks. IPv6 is mandatory, with dual-stack configurations being prevalent. Certain functions demand maximum throughput and transaction rates, necessitating user-plane networking support, such as DPDK. Other functions adhere to conventional cloud-native patterns and can use solutions like OVN-K, kernel networking, and load balancing.

Reviewer: ```suggestion
The networking prerequisites for core telco functions exhibit large diversity, encompassing an array of networking attributes and performance benchmarks. IPv6 is mandatory, with dual-stack configurations being prevalent. Certain functions demand maximum throughput and transaction rates, necessitating user plane networking support, such as DPDK. Other functions adhere to conventional cloud-native patterns and can use solutions like OVN-K, kernel networking, and load balancing.
```
I think "user plane" like "control plane", "data plane".

> Telco core clusters are configured as standard 3-control plane plus N worker nodes configured with the stock non real-time (RT) kernel. In support of workloads with varying networking and performance requirements the worker nodes are segmented into MachineConfigPools for example non-user-dataplane compared to high-throughput nodes. In support of the required telecom operational features the clusters have a standard set of Operator Lifecycle Manager (OLM) day-2 operators installed.

Reviewer: ```suggestion
Telco core clusters are configured as standard three control plane clusters with worker nodes configured with the stock non real-time (RT) kernel. In support of workloads with varying networking and performance requirements, worker nodes are segmented using `MachineConfigPool` CRs. For example, to separate non-user data plane nodes from high-throughput nodes. In support of the required telco operational features, the clusters have a standard set of Operator Lifecycle Manager (OLM) day-2 operators installed.
```
